src/dispatch/plugins/dispatch_slack/bolt.py

```diff
@@ -29,18 +29,19 @@
 
 @app.error
 async def app_error_handler(error, client, body, logger):
-    modal = Modal(
-        title="Error", close="Close", blocks=[Section(text="Something went wrong...")]
-    ).build()
-
     if body:
```

Comment:
I think there can be a situation where an internal app error occurs that isn't related Bolt/Slack and this will not trigger (and result in a silent fail). Going to test and revisit.

Reply:
If possible, we should try and send error messages where the user is. e.g. if a user submits a modal and there is an error, the error should appear in the modal. Sending an ephemeral message in these cases is confusing as the user isn't quite sure which action they took resulted in an error. We can always fall back to ephemeral messages in other when it's not possible.

Reply:
Agreed, it should be easy to check for a `response_url` within the global error handler to mostly get the ideal error flow:

```python
    # the user is in a message flow
    if body.get("response_url"):
        await respond(text=str(error), response_type="ephemeral")
```

One issue w/ this is that Lazy Listener functions don't seem to surface exceptions within them to the error handler at all and also don't currently log the actual exception, it's just generic:

```python
async def handle_add_timeline_submission_event(
    body: dict,
    user: DispatchUser,
    client: AsyncWebClient,
    context: AsyncBoltContext,
    form_data: dict,
):
    """Handles the add timeline submission event."""
    raise Exception("There was an error!")


```

```bash
ERROR:Failed to run an internal function (There was an error!):/Users/wshel/.pyenv/versions/3.9.7/envs/dispatch/lib/python3.9/site-packages/slack_bolt/lazy_listener/async_internals.py:request_wired_wrapper:30
```

![Screenshot 2023-01-10 at 11 06 23 AM](https://user-images.githubusercontent.com/114631109/211639471-8782324a-af1c-4b73-9025-3c1e92ce58c2.png)

When this happens the user will be stuck on the intermediary model created by the `ack_` function and there will be no indication of an issue. Looking for potential workarounds.

Reply:
> One issue w/ this is that Lazy Listener functions don't seem to surface exceptions within them to the error handler at all and also don't currently log the actual exception, it's just generic.

Perhaps we just handle exceptions (e.g. wrap them in try: catch) in lazy functions (instead of relying on the global handler)?